packages/expo-bluetooth/ios/EXBluetooth/EXBluetooth.m

```diff
@@ -0,0 +1,560 @@
+  }
+  return peripheral;
+}
+
+- (CBService *)_getServiceOrReject:(NSString *)uuid peripheral:(CBPeripheral *)peripheral reject:(EXPromiseRejectBlock)reject
+{
+  CBUUID *serviceUUID = [CBUUID UUIDWithString:uuid];
+  CBService *service = [self serviceFromUUID:serviceUUID peripheral:peripheral];
+  if (!service) {
+    reject(@"ERR_NO_SERVICE", [NSString stringWithFormat:@"No valid service with UUID %@ found on peripheral %@", uuid, peripheral.identifier.UUIDString], nil);
+  }
+  return service;
+}
+
+- (CBCharacteristic *)_getCharacteristicOrReject:(NSString *)uuid service:(CBService *)service characteristicProperties:(CBCharacteristicProperties)characteristicProperties reject:(EXPromiseRejectBlock)reject
+{
+  CBUUID *characteristicUUID = [CBUUID UUIDWithString:uuid];
+  
+  CBCharacteristic *characteristic = [self characteristicFromUUID:characteristicUUID service:service prop:characteristicProperties];
+  
+  if (characteristicProperties == CBCharacteristicPropertyNotify && !characteristic) {
+    characteristic = [self characteristicFromUUID:characteristicUUID service:service prop:CBCharacteristicPropertyIndicate];
+  }
+  if (!characteristic) {
+    characteristic = [self characteristicFromUUID:characteristicUUID service:service];
+  }
+  if (!characteristic) {
+    NSString *errorMessage = [NSString stringWithFormat:@"Could not find characteristic with UUID %@ on service with UUID %@ on peripheral with UUID %@",
+                              uuid,
+                              service.UUID.UUIDString,
+                              service.peripheral.identifier.UUIDString];
+    reject(@"ERR_NO_CHARACTERISTIC", errorMessage, nil);
+  }
+  
+  return characteristic;
+}
+
+- (CBDescriptor *)_getDescriptorOrReject:(NSString *)uuid characteristic:(CBCharacteristic *)characteristic reject:(EXPromiseRejectBlock)reject
+{
+  CBDescriptor *descriptor = [self descriptorFromUUID:[CBUUID UUIDWithString:uuid] characteristic:characteristic];
+  if (!descriptor) {
+    NSString *errorMessage = [NSString stringWithFormat:@"Could not find descriptor with UUID %@ on characteristic with UUID %@ on service with UUID %@ on peripheral with UUID %@",
+                              uuid,
+                              characteristic.UUID.UUIDString,
+                              characteristic.service.UUID.UUIDString,
+                              characteristic.service.peripheral.identifier.UUIDString];
+    
+    reject(@"ERR_NO_DESCRIPTOR", errorMessage, nil);
+  }
+  return descriptor;
+}
+
+#pragma mark - Search
+
+-(CBService *)serviceFromUUID:(CBUUID *)UUID peripheral:(CBPeripheral *)peripheral
+{
+  for (CBService *service in peripheral.services) {
+    if ([service.UUID.UUIDString isEqualToString:UUID.UUIDString]) {
+      return service;
+    }
+  }
+  return nil;
+}
+
+- (CBCharacteristic *)characteristicFromUUID:(CBUUID *)UUID service:(CBService *)service prop:(CBCharacteristicProperties)prop
```

Comment:
Write out:

```suggestion
- (CBCharacteristic *)characteristicFromUUID:(CBUUID *)UUID service:(CBService *)service properties:(CBCharacteristicProperties)properties
```

I think it makes sense to document (or rename) whether this method is supposed to return a characteristic with any of the props or all of the props.